conans/test/unittests/util/detected_os_test.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# vim: tabstop=8 expandtab shiftwidth=4 softtabstop=4
+
+import unittest
+import platform
+from mock import mock
+from conans.client.conf.detect import detected_os
+from nose.plugins.attrib import attr
+
+
+class OSInfoTest(unittest.TestCase):
+    @attr('windows')
+    @unittest.skipUnless(platform.system() == "Windows", "Requires windll module")
```

Comment:
Why requiring `windows` if `platform.system` is already mocked? (same for all the other ones below)

Reply:
unfortunately, mocking `platform.system` isn't enough here:
`OSInfo` ctor calls `self.os_version = self.get_win_os_version()` (`oss.py`, line 111)
and it uses ctypes.windll, which is not available on other platforms
not sure how to mock this one

Reply:
on second thought, I probably see a solution how to avoid it - I can mock `get_win_os_version` as well, and `get_linux_distro` too (need to extract it into the function)
will try to implement

Reply:
You can mock it with `with patch.object(OSInfo, "get_win_os_version", return_value="whatever")`.

Reply:
yeah, I've just did it